### PR TITLE
Fix behavior of ice traps in slow chests

### LINKED ISSUE #1073: Ice Trap Chest Speed
Xopar Today at 6:21 AM
A bug cropped up tonight in our rando rando seed. We had "ice_trap_appearance": "anything",, chest cutscenes on, and chest size matches contents on. Ice traps were found in both large and small chests (as expected) however small chests with ice traps still had the chest cutscene enabled. Probably a super easy and quick fix that just fell through the cracks because who plays with chest cutscenes enabled :stuck_out_tongue:

### LINKED ISSUE #1218: Ice Traps ignore Slow Chest Cutscenes setting
When the following settings are used:
Ice Traps appear as Major Items
Chest Size Matches Contents enabled
Fast Chest Cutscenes disabled

Large chests have the long opening animation if the item inside is not an ice trap, but it will have the fast animation if it is an ice trap.

What I expected to happen:
All large chests have the long opening animation, regardless of contents.

### PR DESCRIPTION
**Pull request by fenhl**
The behavior of ice traps with “Fast Chest Cutscenes” off is fixed so that ice traps cloaked as major items use the slow cutscene, while ones cloaked as junk use the fast one. This applies to both a player's own ice traps (major ice traps no longer skip the item get animation if they're in slow chests), as well as other players' ice traps in multiworld, which were previously always slow. As a side effect, ice traps in chests are listed with their cloak models in the spoiler log if “Fast Chest Cutscenes” is off, as was already the case in CAMC.

While investigating this, I also found a bug where plando'd cloak models were applied to the wrong world's item. The fix is included here as well.

Fixes #1073. Fixes #1218.

### CODE REVIEW

**Comment by r0bd0g** [comment]
If you get an ice trap from an NPC or from a "You Are a Fool!" text, you cannot possibly mash out without taking damage, though it's not too difficult to do if the ice trap is from a chest. If chests are going to be giving the "you are a fool" text now, I think it needs to be as easy to mash out without taking damage as it would normally be for a chest. I honestly think this problem needs to be solved or else chests need to not be giving the "you are a fool" text (which doesn't really fix the issue, just passes it further down the road).

**Comment by fenhl** [comment]
I definitely think it makes more sense for major ice traps to give the slow animation like other major items (otherwise you'd know during the animation that it's not an ice trap, or at least not one for yourself). Delaying the damage effect from “fool” ice traps to give players a chance to mash out without taking damage sounds doable but should probably be tracked as a separate issue.

**Comment by r0bd0g** [comment]
I think this change is too much of a problem for players with ice traps and ohko to be accepted as-is.

File: LocationList.py [hunk 1]
```diff
@@ -921,8 +921,9 @@ def shop_address(shop_id, shelf_id):
 }
 
 
-def location_is_viewable(loc_name, correct_chest_appearances):
-    return correct_chest_appearances in ['textures', 'both', 'classic'] and loc_name in location_groups['Chest'] or loc_name in location_groups['CanSee']
+def location_is_viewable(loc_name, correct_chest_appearances, fast_chests):
```
Comment (mracsys) [inline LocationList.py:924]: This is also called in a unit test, update function call there.

**Review by mracsys — COMMENTED** [review]
I feel that being able to mash out of ice trap damage to prevent OHKO deaths goes against the spirit of that combo. It's supposed to be painful. Either way, feels beyond the scope of this PR.

**Comment by r0bd0g** [comment]
Since when is the point of ohko to die over and over again with nothing that you can do about it. You're supposed to try and not die, right? And sometimes that means mashing out of ice traps without taking any damage.

**Comment by fenhl** [comment]
Regardless of the playability impact of this PR without adjusting “fool” ice trap behavior, the two changes are best organized separately. It's clear that you believe this PR should be blocked on that concern, and that can be noted on both descriptions.

**Comment by r0bd0g** [comment]
What I'm saying is we need a 3-step process here. This PR minus adding "you are a fool" to chests. Then, maybe somebody changes how ice traps from sources outside of chests works so that you can mash out. If that happens, then "you are a fool" texts can be added back in.

**Comment by cjohnson57** [comment]
> I think this change is too much of a problem for players with ice traps and ohko to be accepted as-is. (These types of settings would go from kind-of reasonable to completely unplayable immediately.) If you think it's a separate issue, that issue should be addressed first. And if it's too much trouble to be addressed first, then for now you should remove 'you are a fool' from chests.

Wouldn't this only be an issue if ice traps were set to appear as major items? I'd say if you don't want that to happen then don't combine ice traps, OHKO, and ice traps as major items... If you do all 3 settings together then you've kind of asked for it

**Review by cjohnson57 — APPROVED** [review]

**Pull request merged**